pkg/distrofactory/distrofactory.go

```diff
@@ -46,14 +65,48 @@ func (f *Factory) FromHost() distro.Distro {
 }
 
 // New returns a Factory of distro.Distro factories for the given distros.
-func New(factories ...FactoryFunc) *Factory {
-	return &Factory{factories: factories}
+// The provided aliases map has the following constraints:
+// - An alias must not mask an existing distro.
+// - An alias target must map to an existing distro.
+func New(aliases map[string]string, factories ...FactoryFunc) (*Factory, error) {
```

Comment:
TINIEST NITPICK: I think I would prefer if this functionality (registering aliases) was separate.  Having it as the first arg in the constructor makes it stand out too much and while we will definitely use it all the time in composer, I see it as a "secondary" functionality of the distrofactory.

What I'm thinking is basically the following usage pattern:
```go
df := distrofactory.New(fedora.DistroFactory, rhel7.DistroFactory, rhel8.DistroFactory)
err := df.RegisterAliases(aliases)
```

or also from the default level:
```go
df := distrofactory.NewDefault()
err := df.RegisterAliases(aliases)
```

I'm fine to keep it as is.  Change it only if you like it :)

Reply:
So, I'm not opposed to the suggestion. I'll have a closer look and consider making the change.

Reply:
I made the change. I think that it is a bit nicer in all the places that do not set the alias and that one does not need to check the error...